src/dagParser.cpp

```diff
@@ -0,0 +1,91 @@
+#include "ndn-cxx/util/logger.hpp"
+
+#include "dagParser.hpp"
+
+namespace iceflow {
+
+NDN_LOG_INIT(iceflow.DAGParser);
+
+DAGParser DAGParser::parseFromFile(const std::string &filename) {
+  DAGParser app;
+  std::ifstream file(filename);
+  if (!file.is_open()) {
+    throw std::runtime_error("File can not be opened " + filename);
+  }
+
+  json dagParam;
+  file >> dagParam;
+
+  // Parsing the application name
+  app.applicationName = dagParam.at("applicationName").get<std::string>();
```

Comment:
Sorry if I am commenting prematurely on this, but maybe both the `applicationName` and the `nodes` could rather be turned into constructor parameters.